With React 19 typings: Cannot find namespace 'JSX'.

## Bug report

### Current Behavior

I use the latest formik release with React v19, specifically `"@types/react": "^19.0.1"` and I get 4 errors like:

```
node_modules/formik/dist/types.d.ts:209:109 - error TS2503: Cannot find namespace 'JSX'.

209 export type GenericFieldHTMLAttributes = JSX.IntrinsicElements['input'] | JSX.IntrinsicElements['select'] | JSX.IntrinsicElements['textarea'];
```

### Expected behavior

No type errors.

### Suggested solution(s)

`s/JSX/React.JSX`

### Additional context

https://react.dev/blog/2024/04/25/react-19-upgrade-guide#the-jsx-namespace-in-typescript

It's only about typings anyway. In `@react/types`, this was backported also to the v16 series, so the proposed change doesn't require React v19 to work.

However, it _does_ require React, and currently formik doesn't declare that even as a peer dependency, which surprises me.

### Your environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Formik           | 2.4.6
| React            | 19.0.0
| TypeScript       | 5.5.4
| npm/Yarn         | npm 10.9.0
| Operating System | Kubuntu

--- a/packages/formik/src/types.tsx
+++ b/packages/formik/src/types.tsx
@@ -279,9 +279,9 @@ export interface SharedRenderProps<T> {
 }
 
 export type GenericFieldHTMLAttributes =
-  | JSX.IntrinsicElements['input']
-  | JSX.IntrinsicElements['select']
-  | JSX.IntrinsicElements['textarea'];
+  | React.JSX.IntrinsicElements['input']
+  | React.JSX.IntrinsicElements['select']
+  | React.JSX.IntrinsicElements['textarea'];
 
 /** Field metadata */
 export interface FieldMetaProps<Value> {

--- a/website/src/components/Container.tsx
+++ b/website/src/components/Container.tsx
@@ -2,7 +2,7 @@ import * as React from 'react';
 import cn from 'classnames';
 import { HTMLProps } from 'react';
 
-export const Container: React.FC<JSX.IntrinsicElements['div']> = ({
+export const Container: React.FC<React.JSX.IntrinsicElements['div']> = ({
   className,
   ...props
 }) => {

--- a/website/src/components/FormiumLogo.tsx
+++ b/website/src/components/FormiumLogo.tsx
@@ -1,6 +1,6 @@
 import * as React from 'react';
 
-export const FormiumLogo: React.FC<JSX.IntrinsicElements['svg']> = props => {
+export const FormiumLogo: React.FC<React.JSX.IntrinsicElements['svg']> = props => {
   return (
     <svg
       width={'101px'}

--- a/website/src/components/forwardRefWithAs.tsx
+++ b/website/src/components/forwardRefWithAs.tsx
@@ -96,7 +96,7 @@ export function forwardRefWithAs<Props, ComponentType extends As>(
 Test components to make sure our dynamic As prop components work as intended 
 type PopupProps = {
   lol: string;
-  children?: React.ReactNode | ((value?: number) => JSX.Element);
+  children?: React.ReactNode | ((value?: number) => React.JSX.Element);
 };
 export const Popup = forwardRefWithAs<PopupProps, 'input'>(
   ({ as: Comp = 'input', lol, className, children, ...props }, ref) => {

